modules/bastion/main.tf

```diff
@@ -34,20 +32,11 @@ locals {
   })
 }
 
-data "aws_route53_zone" "vanity" {
-  count = local.route53_enabled ? 1 : 0
+data "aws_route53_zone" "route53_zone" {
+  count = try(length(var.vanity_domain), 0) > 0 ? 1 : 0
   name  = var.vanity_domain
```

Comment:
The vanity domain should be renamed to something like `var.vanity_domain_zone` or something similar.

Reply:
Vanity domain is how the customer chose to use it. But I think it's too opinionated. Let's just call it dns zone.

Reply:
See our route 53 module. Use that instead.

Reply:
@osterman 

> See our route 53 module. Use that instead.

Do you mean the variable description in that module or the module itself? I'm assuming the former, because the `ec2-bastion-server` module manages Route53 resources itself.

Reply:
resolved